### PR TITLE
tsp, fix encoding issue for special chars

### PR DESCRIPTION
**Pull request by haolingdong-msft**
Link https://github.com/Azure/autorest.java/issues/2186

Fix the wrong generated code for special chars.

got failure like below when generating loadingtesting:

```java
    /*
     * The aggregation function to be applied on the client metric. Allowed functions
     * - ‘percentage�? - for error metric , ‘avg�?, ‘p50�?, ‘p90�?, ‘p95�?, ‘p99�?, ‘min�?,
     * ‘max�? - for response_time_ms and latency metric, ‘avg�? - for requests_per_sec,
     * ‘count�? - for requests
     */
    @Generated
    @JsonProperty(value = "aggregate")
    private PfAgFunc aggregate;
source here https://github.com/Azure/azure-rest-api-specs/blob/feature/loadtesting/specification/loadtestservice/models.tsp#L209-L214
```

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
see https://github.com/Azure/autorest.java/issues/2186#issuecomment-1584164215

better not to rely on `file.encoding` env var.

**Comment by haolingdong-msft** [comment]
> see [#2186 (comment)](https://github.com/Azure/autorest.java/issues/2186#issuecomment-1584164215)
> 
> better not to rely on `file.encoding` env var.

ok, I will try.

**Comment by weidongxu-microsoft** [comment]
writer part may have bug as well.
```java
        try (FileWriter writer = new FileWriter(outputFile)) {
            writer.write(content);
        } catch (IOException e) {
```

you may need to write this
```java
        try (OutputStreamWriter writer = new OutputStreamWriter(Files.newOutputStream(outputFile.toPath()), StandardCharsets.UTF_8)) {
            writer.write(content);
        } catch (IOException e) {
```

File: typespec-extension/src/emitter.ts [hunk 1]
```diff
@@ -100,6 +100,7 @@ export async function $onEmit(context: EmitContext<EmitterOptions>) {
     try {
       const output = await promisify(execFile)("java", [
         `-DemitterOptions=${emitterOptions}`,
+        `-Dfile.encoding=UTF-8`,
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/emitter.ts:103]: let's see if it will pass after remove this.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
make sure you always use latest main when create branch

**Pull request merged**